### DISCUSSION TITLE
How to allow automatic installation of LSP servers in headless mode?

### DISCUSSION THREAD

**Discussion by amitds1997** [mason-org/mason.nvim#1590] [Q&A]
I maintain a plugin: [remote-nvim.nvim](https://github.com/amitds1997/remote-nvim.nvim) that provides VS Code-like remote development features in Neovim. To do this, I run Neovim in `headless` mode. And then use the [`--remote-ui` feature](https://neovim.io/doc/user/remote.html#--remote-ui) to connect to the remote headless Neovim server. Until a few weeks back, the auto installation used to work, and now it does not. I do see this was merged [way before](https://github.com/williamboman/mason-lspconfig.nvim/pull/175) so I'm not even sure how it used to work till now.

Is it possible to configure mason-lspconfig to allow auto installation to work again in headless mode?

**Comment by kureta** [comment]
I found a hacky way and explained it [here](https://github.com/LazyVim/LazyVim/discussions/3679#discussioncomment-10368889)

**Comment by william-barros-costa** [comment]
I have not tested this but this tool seems to offer what you both need @kureta and @amitds1997 
https://github.com/WhoIsSethDaniel/mason-tool-installer.nvim?tab=readme-ov-file#commands (See :MasonToolsInstallSync)
